Quantum chess bug fixes

losos0 opened the pull request:
1. Fix equality test for `Move`
2. In the interactive board, don't log measurement statistics about the initial position
3. Fix a bug where a qubit with only single-qubit gates can be mapped to multiple grid qubits

dstrain115 reviewed (commented):
Can you add a description to the PR to explain the bugs?  I'm not totally sure I understand what the problem was.

Also, I don't see any regression tests that test them.  The usual practice is to put a failing test that is fixed by the change.

losos0 commented:
Added description to the OP.

- (1) already has a test.
- (2) is about logging debugging information so there is no test. I added additional description to the commit message to clarify.
- Added a test for (3).

dstrain115 commented on recirq/quantum_chess/circuit_transformer.py:
```diff
@@ -323,8 +319,7 @@ def qubit_mapping(self,
 
-        if len(mapping) != len(g):
-            print('Warning: could not map all qubits!')
+        assert len(mapping) == len(g) + len(sq)
```
Should we assert with a message so that people get a message?
Alternatively, should this raise an appropriate Exception instead?

losos0 replied:
There is always supposed to be a one-to-one mapping between named qubits and allocated grid qubits. An assertion seems appropriate to me since if it does not hold it means there is a bug in qubit_mapping.

Added a message to the assertion

dstrain115 reviewed (approved):
LGTM after a minor comment.
